src/Yoti.Auth/ShareUrl/Policy/DynamicPolicyBuilder.cs

```diff
@@ -13,100 +13,108 @@ public DynamicPolicyBuilder WithWantedAttribute(WantedAttribute wantedAttribute)
             Validation.NotNull(wantedAttribute, nameof(wantedAttribute));
 
             string key = wantedAttribute.Derivation ?? wantedAttribute.Name;
+
+            if (wantedAttribute.Constraints?.Count > 0)
+            {
+                key += wantedAttribute.Constraints.GetHashCode();
```

Comment:
I think I changed this in Java to be separated by a dash so it's just a bit easier on the eyes to debug if ever needed (i.e. `document_images-48927` rather than `document_images48927`)